crates/frontend/src/pages/dimensions.rs

```diff
@@ -32,10 +35,30 @@ pub fn dimensions() -> impl IntoView {
         },
     );
 
+    let confirm_delete = Callback::new(move |_| {
+        if let Some(id) = delete_id.get().clone() {
+            spawn_local(async move {
```

Comment:
when an api call is going on, shouldn't the interactions be totally blocked here ?
if it fails or succeeds the user will totally lose context, they could have to some other section assuming that the operation (api call) happened instantaneously and that too it was a success